leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/LwM2mQueue.java

```diff
@@ -0,0 +1,200 @@
+        this.clientAwakeTime = clientAwakeTime; /* ms */
+    }
+
+    /* Client State Control */
+
+    /**
+     * Set the client state to awake. This should be called when an update message is received from the server. It
+     * starts the client awake timer.
+     */
+    public void setAwake() {
+        if (state == Presence.SLEEPING) {
+            state = Presence.AWAKE;
+        }
+        startClientAwakeTimer();
+    }
+
+    /**
+     * Set the client state to sleeping. This should be called when the the time the client waits before going to sleep
+     * expires, or when the client is not responding. It also notifies the listeners inside the {@link PresenceService}.
+     * It stops the client awake timer.
+     */
+    public void setSleeping() {
+        if (state == Presence.AWAKE) {
+            state = Presence.SLEEPING;
+        }
+        stopClientAwakeTimer();
+        presenceService.notifySleeping(registration);
+    }
+
+    /**
+     * Tells if the client is sleeping or not
+     * 
+     * @return true if client is sleeping
+     */
+    public boolean isClientSleeping() {
+        return state == Presence.SLEEPING;
+    }
+
+    /* Control of the time the client waits before going to sleep */
+    /**
+     * Get the time that the client stays awake after an update message or the last received request.
+     * 
+     * @return The client awake time.
+     */
+    public int getClientAwakeTime() {
+        return clientAwakeTime;
+    }
+
+    /**
+     * Sets the client awake time, in case it wants to be modified during run time.
+     * 
+     * @param clientAwakeTime
+     */
+    public void setClientAwakeTime(int clientAwakeTime) {
+        this.clientAwakeTime = clientAwakeTime;
+    }
+
+    /**
+     * Start or restart (if already started) the timer that handles the client wait before sleep time.
+     */
+    public void startClientAwakeTimer() {
+
+        if (clientAwakeTime != 0) {
+            if (clientAwakeTimer != null) {
+                clientAwakeTimer.cancel();
+                clientAwakeTimer.purge();
+            }
+            clientAwakeTimer = new Timer();
+            clientAwakeTask = new TimerTask() {
+
+                @Override
+                public void run() {
+                    if (!isClientSleeping()) {
+                        setSleeping();
+                    }
+                }
+            };
+            clientAwakeTimer.schedule(clientAwakeTask, clientAwakeTime);
+        }
+
+    }
+
+    /**
+     * Stop the timer that handles the client wait before sleep time.
+     */
+    private void stopClientAwakeTimer() {
+
+        if (clientAwakeTimer != null) {
+            clientAwakeTimer.cancel();
+            clientAwakeTimer.purge();
+        }
+
+    }
+
+    /**
+     * Reads from the Californium.properties file the MAX_TRANSMIT_WAIT constant of the CoAP Protocol
+     * 
+     * @return MAX_TRANSMIT_WAIT
+     */
+    public int readCoapMaxTransmitWait() {
```

Comment:
Californium.properties is californium stuff and `leshan-server-core` does not know Californium.
This introduce an implicit dependency. 

We should replace this by a parameter or a Constant.